src/lib/get-html.ts

```diff
@@ -1,14 +1,67 @@
+import marked from 'marked';
 import { Config } from './config';
-import { getMarked } from './get-marked-with-highlighter';
+import { getHighlightRenderer } from './get-marked-with-highlighter';
+import { getHeadingRenderer, TableOfContent, toc } from './get-marked-with-toc';
 
 /**
  * Generates a HTML document from a markdown string and returns it as a string.
  */
-export const getHtml = (md: string, config: Config) => `<!DOCTYPE html>
-<html>
-<head><meta charset="utf-8"></head>
-<body class="${config.body_class.join(' ')}">
-${getMarked(config.marked_options)(md)}
-</body>
-</html>
-`;
+export const getHtml = (md: string, config: Config) => {
+	marked.setOptions(config.marked_options);
+
+	(marked as any).use({ renderer: getHighlightRenderer(config.marked_options) });
+	(marked as any).use({ renderer: getHeadingRenderer(config).renderer });
+
+	const markedContent: string = marked(md);
+	let html = `<!DOCTYPE html>
+	<html>
+		<head><meta charset="utf-8"></head>
+		<body class="${config.body_class.join(' ')}">
+			${markedContent}
+		</body>
+	</html>
+	`;
+
+	// generate table of content only if we have headings
+	// const { toc } = getHeadingRenderer(config);
+	if (html.includes('<!-- TOC -->') && toc.length !== 0) {
+		const ctx = [];
```

Comment:
What does `ctx` stand for?

Reply:
stand for context, but something like toc_builder or generated_toc may be more appropriate?

Reply:
How about `parts`